src/plugins/discover/public/application/view_alert/view_alert_route.tsx

```diff
@@ -0,0 +1,187 @@
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+import React, { useEffect, useMemo } from 'react';
+import { useHistory, useLocation, useParams } from 'react-router-dom';
+import { sha256 } from 'js-sha256';
+import { i18n } from '@kbn/i18n';
+import { ToastsStart } from 'kibana/public';
+import type { Alert } from '../../../../../../x-pack/plugins/alerting/common';
+import { AlertTypeParams } from '../../../../../../x-pack/plugins/alerting/common';
+import { getTime, SerializedSearchSourceFields } from '../../../../data/common';
+import type { Filter, TimeRange } from '../../../../data/public';
+import { MarkdownSimple, toMountPoint } from '../../../../kibana_react/public';
+import { DiscoverAppLocatorParams } from '../../locator';
+import { useDiscoverServices } from '../../utils/use_discover_services';
+
+interface SearchThresholdAlertParams extends AlertTypeParams {
+  searchConfiguration: SerializedSearchSourceFields;
+}
+
+interface QueryParams {
+  from: string | null;
+  to: string | null;
+  checksum: string | null;
+}
+
+type NonNullableEntry<T> = { [K in keyof T]: NonNullable<T[keyof T]> };
+
+const LEGACY_BASE_ALERT_API_PATH = '/api/alerts';
+const DISCOVER_MAIN_ROUTE = '/';
+
+const displayError = (title: string, errorMessage: string, toastNotifications: ToastsStart) => {
+  toastNotifications.addDanger({
+    title,
+    text: toMountPoint(<MarkdownSimple>{errorMessage}</MarkdownSimple>),
+  });
+};
+
+const displayRuleChangedWarn = (toastNotifications: ToastsStart) => {
+  const warnTitle = i18n.translate('discover.viewAlert.alertRuleChangedWarnTitle', {
+    defaultMessage: 'Alert rule has changed',
+  });
+  const warnDescription = i18n.translate('discover.viewAlert.alertRuleChangedWarnDescription', {
+    defaultMessage: `Displayed documents might not match the documents triggered notification, 
+    since the rule configuration has been changed.`,
+  });
+
+  toastNotifications.addWarning({
+    title: warnTitle,
+    text: toMountPoint(<MarkdownSimple>{warnDescription}</MarkdownSimple>),
+  });
+};
+
+const getCurrentChecksum = (params: SearchThresholdAlertParams) =>
+  sha256.create().update(JSON.stringify(params)).hex();
+
+const isConcreteAlert = (
+  queryParams: QueryParams
+): queryParams is NonNullableEntry<QueryParams> => {
+  return Boolean(queryParams.from && queryParams.to && queryParams.checksum);
+};
+
+export function ViewAlertRoute() {
+  const { core, data, locator, toastNotifications } = useDiscoverServices();
+  const { id } = useParams<{ id: string }>();
+  const history = useHistory();
+  const { search } = useLocation();
+
+  const query = useMemo(() => new URLSearchParams(search), [search]);
+
+  const queryParams: QueryParams = useMemo(
+    () => ({
+      from: query.get('from'),
+      to: query.get('to'),
+      checksum: query.get('checksum'),
+    }),
+    [query]
+  );
+
+  const openConcreteAlert = isConcreteAlert(queryParams);
```

Comment:
wouldn't `isValidAlert` a slightly better wording here (was not sure about concrete in English ... there's a similar German word, but my first thought was concrete like cement 😄 )

Reply:
we synced offline and @Dmitriynj explained that this route is also used for the 'View in App' button ... makes sense, also we talked about that `isActualAlert` could be a better naming in this case